routes/dbapi.js

```diff
@@ -0,0 +1,60 @@
+const MongoClient = require('mongodb').MongoClient;
+const autoIncrement = require('mongodb-autoincrement');
+const dbUrl = 'mongodb://localhost:27017/guess';
+
+let _db;
+
+/**
+ * MongoDB operations interface
+ */
+function db() {
+  MongoClient.connect(dbUrl, function(err, database) {
+    if (err) return console.log(err);
+    _db = database;
+  });
+}
+
+db.prototype.insertInCollection = function(collection, payload, callback) {
+	// validate(req.body)
+	let collec = _db.collection(collection);
+	// Check if word already exists
+	collec.find({word: payload.word}).toArray((err, results) => {
+		if (err) return callback(this.handleInsertionError(err));
+		if (results.length) return callback('Duplicate');
+
+		autoIncrement.getNextSequence(_db, collection, (err, autoIndex) => {
+			// Get auto incremented Id
+			payload._id = autoIndex;
+			collec.insert(payload, (err, result) => {
+				if (err) return callback(this.handleInsertionError(err));
+        return callback('Saved');
+			});
+		});
+	});
+};
+
+db.prototype.retrieveGamesFollowingId = function(collection, id, callback) {
+	this.readCollection(collection, (results) => {
+		let response = [];
+		for(let i in results) {
```

Comment:
consider a scenerio of 10k games where id given is 999, this would become pretty slow, can't we form a query for this?